### PR TITLE
Rewrite JS hooks

### PR DESCRIPTION
**Pull request by MrPetovan**
- Use event listeners instead of homebrew hooks
- Remove view/js/addon-hooks.js and its references
- Update Addon docs

### CODE REVIEW

**Comment by MrPetovan** [comment]
I found out that the `App->page['end']` content string was never included in any templates. Good riddance.

Instead, I'm leveraging the `App->page['footer']` part to register deferred Javascript files from addons.

I also added a new `head` hook to register stylesheets and remove even more HTML writing in the addon PHP files.

File: src/App.php [hunk 1]
```diff
@@ -96,6 +96,23 @@ class App
 	public $force_max_items = 0;
 	public $theme_events_in_profile = true;
 
+	public $stylesheets = [];
+	public $footerScripts = [];
+
+	public function registerStylesheet($path)
```
Comment (rabuzarus) [inline src/App.php]: please add a doxygen header

File: src/App.php [hunk 2]
```diff
@@ -96,6 +96,23 @@ class App
...
+	{
+		$url = str_replace($this->get_basepath() . DIRECTORY_SEPARATOR, '', $path);
+
+		$this->stylesheets[] = trim($url, '/');
+	}
+
+	public function registerFooterScript($path)
```
Comment (rabuzarus) [inline src/App.php]: please add a doxygen header

File: src/App.php [hunk 3]
```diff
@@ -797,20 +812,51 @@ public function init_pagehead()
 			'$shortcut_icon'   => $shortcut_icon,
 			'$touch_icon'      => $touch_icon,
 			'$stylesheet'      => $stylesheet,
```
Comment (rabuzarus) [inline src/App.php]: shouldn't be `$stylesheet` the first entry of `$stylesheets`?

File: src/App.php [hunk 4]
```diff
@@ -733,7 +749,7 @@ public function set_pager_page($n)
 		$this->pager['start'] = ($this->pager['page'] * $this->pager['itemspage']) - $this->pager['itemspage'];
 	}
 
-	public function init_pagehead()
+	public function initHead()
```
Comment (rabuzarus) [inline src/App.php]: please add a doxygen header

**Comment by rabuzarus** [comment]
> I found out that the App->page['end'] content string was never included in any template

Frost and Frost-Mobile used it (https://github.com/friendica/friendica/blob/3.5.1/view/theme/frost/php/default.php#L41). But this themes were removed because they were deprecated.

**Comment by MrPetovan** [comment]
Ah, thanks, it wasn't completely pointless then.

File: src/App.php (hunk 3 above)
Comment (MrPetovan) [reply src/App.php]: I don't understand exactly what the `$stylesheet` template variable is supposed to hold, but one thing for sure is that it doesn't hold any URL that I can figure out.

**Comment by MrPetovan** [comment]
Can someone please review this? I'm planning on working further on `src/App.php`, but my server currently runs this branch and I'd rather start from develop.

**Comment by rabuzarus** [comment]
From my side I give a OK. @annando can you have a look please

**Comment by annando** [comment]
This is so much frontend stuff, that I trust @rabuzarus, I guess.

@MrPetovan sorry, I hadn't seen this request, I had been too busy with the other stuff.

**Comment by MrPetovan** [comment]
The `addon_hooks` stuff was only used for the Mathjax addon. I've ben running this branch non-stop on my server for 12 days now, and it is blocking me for further work.

**Pull request merged**